articles/golang-part-2-programming-basics/index.md

```diff
@@ -0,0 +1,304 @@
+
+#### Directories
+A workspace is a directory hierarchy with two directories at its root: src, bin, and pkg.
+
+**src**: *The src directory contains the source code (Go programs) written by the user. It may contain multiple repositories*.
+
+**bin**: *This folder contains the binary executable files that are built and installed by the Go tool*.
+
+**pkg**: *This directory contains all the non-executable packages or shared libraries that are used in other executable packages*.
+
+This is an example of a workspace that contains an executable program called hello inside the bin directory and multiple programs inside the src directory in a hierarchical structure.
+
+```
+bin/
+    hello                                                   
+src/
+    github.com/golang/example/                     
+        hello/
+            hello.go
+        program/
+            program.go               
+```
+
+### Program structure
+Go programs have the following structure:
+
+```go
+package main
+
+import "fmt"
+
+func main() {
+
+   fmt.Println("Hello, World!")
+}
+```
+
+1. Package declaration: *This is The first line of the program and defines the name of the current package name. "Package main", tells Go that this is the starting point to run the program. We can use the **package** keyword to define packages in our program*.
+
+2. Imports: *Import statements tell the Go compiler to include the files defined in that particular package. In the above example, we are importing the "fmt" package that implements I/O functions in Go. We can use the **import** keyword to import a particular package to our program*.   
+
+3. Functions: *A function is a set of statements that perform a particular task. The main function defines where the program begins and ends. The main function is mandatory to define in the main package. Go automatically executes everything in the main function, and the program exits when it reaches the end. The **func** keyword is used to define functions*.
+
+The "go run" command can be used to run the go programs. This command compiles and executes the program on the terminal or command prompt. Copy and paste all the examples in this tutorial onto a file, give the file a name, and run it using the "go run <file name>" command on your terminal.
+
+![go run](/engineering-education/golang-part-2-programming-basics/go-run.png)
+
+#### Variables, Keywords, and identifiers
+**Identifiers** are names of various components defined by the user in the program. An identifier can be the name of a variable, function, constant, statement, etc. **Keywords** are words that are reserved by a program because they have a special meaning and are used for some internal action. Keywords are reserved and cannot be used as identifiers in a program. There are 25 keywords in Go:
+
+![go keywords](/engineering-education/golang-part-2-programming-basics/go-keywords.png)
```

Comment:
Would this work better as a markdown table rather than an image? 
This will allow users to copy-paste the keywords. 

Not a blocker to releasing the article though. Feel free to keep the image.